### PR TITLE
Support cluster bootstrapping in vtctldclient

### PR DESCRIPTION
**Pull request by mattlord**
## Description

This meets the one use case that remained for the `vtctl` client — which was a combination of `vtctlclient` and `vtctld` — bootstrapping a new cluster by creating the first cell that you can then start a `vtctld` server in.

More generally, this PR adds support to `vtctldclient` so that you can execute any command w/o requiring a running `vtctld` server by specifying `--server internal`. When that value is specified, `vtctldclient` acts as the legacy `vtctl` client in that it uses an internal `VtctldServer` implementation — interfacing with the topology server(s) directly.

## Related Issue(s)

  - Fixes: https://github.com/vitessio/vitess/issues/14317

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by ajm188** [comment]
> This PR adds support to vtctldclient so that you can execute any command w/o requiring a running vtctld server by specifying --server internal

My thinking instead is that if a `vtctld` starts, it first attempts to create the root `CellInfo` object if it doesn't already exist (and so there's no need to do this separately in the client)

**Comment by mattlord** [comment]
> > This PR adds support to vtctldclient so that you can execute any command w/o requiring a running vtctld server by specifying --server internal
> 
> My thinking instead is that if a `vtctld` starts, it first attempts to create the root `CellInfo` object if it doesn't already exist (and so there's no need to do this separately in the client)

I don't know that I'm against this idea... but the current implementation here also generally provides a nice replacement for the`vtctl` client/binary as a whole.

**Comment by ajm188** [comment]
right but if we can put this in `vtctld` we can drop the `vtctl` binary entirely

**Comment by ajm188** [comment]
> as vtctldclient --server internal would be the equivalent.

you're completely right! ignore me ... 🙈

File: go/flags/endtoend/vtctldclient.txt [hunk 1]
```diff
@@ -1,4 +1,7 @@
-Executes a cluster management command on the remote vtctld server.
+If there are no running vtctld servers -- for example when bootstrapping
```
Comment (ajm188) [inline go/flags/endtoend/vtctldclient.txt:1]: something is wrong here

File: go/cmd/vtctldclient/command/root.go [hunk 2]
```diff
@@ -54,14 +68,36 @@ var (
...
+		implementation:        "etcd2",
+		globalServerAddresses: []string{"localhost:2379"},
+		globalRoot:            "/vitess/global",
+	}
+
 	// Root is the main entrypoint to the vtctldclient CLI.
 	Root = &cobra.Command{
 		Use:   "vtctldclient",
 		Short: "Executes a cluster management command on the remote vtctld server.",
+		Long: fmt.Sprintf(`If there are no running vtctld servers -- for example when bootstrapping
```
Comment (ajm188) [inline go/cmd/vtctldclient/command/root.go:97]: ah, we need to keep the `Short` as the first sentence here (imo)

**Review by ajm188 — APPROVED** [review]
one help text change but lgtm

File: go/cmd/vtctldclient/command/root.go (hunk 2 above)
Comment (mattlord) [reply go/cmd/vtctldclient/command/root.go:97]: Good catch! I didn't pay too much attention to the help text output and didn't realize the short was then not shown.

**Review by frouioui — APPROVED** [review]

**Comment by frouioui** [comment]
Hey @mattlord do we still want to merge this?

**Comment by ajm188** [comment]
> Hey @mattlord do we still want to merge this?

yes, we do!

**Comment by mattlord** [comment]
I will get this fixed up and merged soon. I will add a commit here to address this non-behavior changing discussion as well before merging: https://github.com/vitessio/vitess/pull/14467#pullrequestreview-1732156124

**Pull request merged**